[MM-23135] Remove double-handling of toggle switch state

chuttam opened the pull request:
#### Summary

Whether the Mute, Favorite, Ignore toggles are set on or off is already being governed by the parent controlling component. Whenever any of their respective states change, the parent updates the child's props, and the toggle switches re-render with the correct switch settings in props anyway.

`getDerivedStateFromProps` is called on component creation and then whenever the parent component's state is updated and a re-render of all children is triggered, **regardless of whether the props are different from before **

The `this.setState` in each of the toggle switch handlers was already taking care of keeping component state healthy. With `getDerivedStateFromProps`, state was being "double-managed" resulting in toggle switches sometimes jumping between on and off, well after it was touched.

#### Ticket Link

* https://mattermost.atlassian.net/browse/MM-23135

#### Device Information
This PR was tested on: Android 10 (OnePlus 5 device)

#### Screenshots

**Old (jumping switches)**
<img width="275" alt="old" src="https://user-images.githubusercontent.com/887849/76338796-f66c3a00-62d7-11ea-8d83-5b63d0f3d01c.gif">

**New**

<img width="275" alt="new" src="https://user-images.githubusercontent.com/887849/76338784-f3714980-62d7-11ea-86df-c3bb6030f067.gif">

enahum commented:
@chuttam what happens if looking at the screen you change the prop on the webapp ?

chuttam commented:
> @chuttam what happens if looking at the screen you change the prop on the webapp ?

The toggle on the mobile app screen will not change, unless the app is reloaded, which is the existing behaviour on `master` at the moment too.

enahum commented:
@chuttam I can see it updating on the master branch.

chuttam commented:
Websocket issues when testing agains Community. Switched to local server, and I was able to see the real-time screen update when changing toggles on web app.

Component state initialization on re-render is now adjusted. Back to review @enahum

mattermod commented:
Building app in separate branch.

mattermod commented:
Successfully building:  
https://circleci.com/gh/mattermost/mattermost-mobile/4230  
https://circleci.com/gh/mattermost/mattermost-mobile/4229

mattermod commented:
Artifact links:  
https://4230-70265724-gh.circle-artifacts.com/0/%7E/mattermost-mobile/Mattermost_Beta.ipa  
https://4229-70265724-gh.circle-artifacts.com/0/%7E/mattermost-mobile/Mattermost_Beta.apk

chuttam commented:
I'm definitely seeing the bad behaviour worsen on an iPhone 11. Android is behaving much better though.

Definitely more going on here than meets the eye. I'll dive in first chance I get.

chuttam commented:
After analysis, ticket [MM-22582](https://mattermost.atlassian.net/browse/MM-22582) is open for the possible underlying issue. Until this root cause is resolved, this PR is moot, although the `setState` calls should be removed at that point too, because they cause even more state noise.

Closing.
